execution-core/src/transfer.rs

```diff
@@ -328,6 +328,10 @@ pub struct ContractDeploy {
     pub owner: Vec<u8>,
     /// Constructor arguments of the deployed contract.
     pub constructor_args: Option<Vec<u8>>,
+    /// Nonce for contract id uniqueness
+    pub nonce: u64,
+    /// Metadata
+    pub metadata: Vec<u8>,
```

Comment:
Metadata may contain name, version, authors, build information. I think we need it as only bytecode and nonce do not give us variability in the sense of contract id. What if we want the same contract bytecode deployed several times. Changing nonce for that would be a workaround, but proper solution is to have a metadata field which explicitly introduces the required information.

Reply:
I see. However, with this implementation is is used for none of those things, since it is unconstrained. If the goal is to have more variability, then simply increasing the width of the nonce would suffice.
A different nonce is not a workaround BTW. It is in line with how nonces are used everywhere else - number that's used once.

Reply:
I agree with @ureeves on this one. I would like to only introduce extra-fields when we really need them. And to me the `nonce` is more than enough for our current requirements (the need to produce a unique contract-id).

Reply:
Nonce used for various purposes is suboptimal IMO. If we use nonce for vanity, we should not use the same nonce for other kinds of variability IMO. Nevertheless, I removed the metadata as we agreed that nonce, bytecode and owner are sufficient for variability.